File: StoreAPI/StoreAPI/Models/Store.cs
namespace StoreAPI.Models
{
    public enum StoreCategory
    {
        General,
        Food,
        Clothing,
        Electronics,
        Furniture
    }

    public class Store
    {
        public virtual long Id { get; set; }
        public virtual string? Name { get; set; }
        public virtual string? Description { get; set; }

        public virtual StoreCategory Category { get; set; }
        public virtual string? Address { get; set; }

        public virtual string? City { get; set; }
        public virtual string? State { get; set; }

        // ZIP codes can start with 0, unfortunately
        public virtual string? ZipCode { get; set; }
        public virtual string? Country { get; set; }

        public virtual DateTime? OpenDate { get; set; }
        public virtual DateTime? CloseDate { get; set; }

        // Hidden from the API because it's not in the DTO
        public virtual ICollection<StoreShift> StoreShifts { get; set; } = null!;

        public virtual long? UserId { get; set; }
        public virtual User? User { get; set; } = null!;
    }
}

File: StoreManagement/src/models/Store.ts
import { StoreShift } from "./StoreShift";
import { User } from "./User";

export enum StoreCategory {
    General,
    Food,
    Clothing,
    Electronics,
    Furniture,
}

export interface Store {
    id?: number;
    name: string;
    description: string;

    category: StoreCategory;
    address: string;

    city: string;
    state: string;

    zipCode: string;
    country: string;

    openDate?: string;
    closeDate?: string;

    storeShifts?: StoreShift[];

    userId?: number;
    user?: User;
}
